dev/src/transaction.ts

```diff
@@ -460,13 +492,14 @@ export class Transaction implements firestore.Transaction {
    * @private
    * @internal
    */
-  commit(): Promise<void> {
-    return this._writeBatch
-      ._commit({
-        transactionId: this._transactionId,
-        requestTag: this._requestTag,
-      })
-      .then(() => {});
+  async commit(): Promise<void> {
+    if (this._readTime) {
```

Comment:
ReadOnly but without ReadTime are still transactions that have `BeginTransaction` and `Commit` request. That is why I have to use `_readTime`, since only in the subset of cases where the transaction if ReadOnly AND with ReadTime can we do the optimization.

However, this is also a little weird of a check that I added, since this will only occur if there is a bug in SDK. The customer never has an opportunity to call `commit()` on transactions. Calling `commit()` is done implicitly when update lambda completes. So this should never happen so long as our SDK logic is correct.